### PR TITLE
Add support for Fat MachO files in the minidump-unwind debuginfo symbol provider

### PR DESCRIPTION
**Pull request by afranchuk**
I have not tested this yet, but it should do the trick. I will test it out through `minidump-stackwalk`.

### CODE REVIEW

File: minidump-unwind/src/symbols/debuginfo.rs [hunk 1]
```diff
@@ -424,6 +449,39 @@ fn load_unwind_module(module: &dyn Module) -> Option<(Mmap, framehop::Module<Mod
...
+            match arch.data(fatfile_data) {
+                Err(e) => {
+                    tracing::error!(
+                        "failed to read data from fat macho file {}: {e}",
+                        path.display()
+                    );
+                    None
+                }
+                Ok(data) => Some(data),
+            }
```
Comment (gabrielesvelto) [inline minidump-unwind/src/symbols/debuginfo.rs:480]: You can probable use `map_or_else()` here.

**Review by gabrielesvelto — APPROVED** [review]
I left just a small nit, the rest is fine.

File: minidump-unwind/src/symbols/debuginfo.rs (hunk 1 above)
Comment (afranchuk) [reply minidump-unwind/src/symbols/debuginfo.rs:480]: That's one I've needed to reach for infrequently enough that I forget about it :smile:

**Review by gabrielesvelto — APPROVED** [review]

**Comment by afranchuk** [comment]
I have verified this works.

**Pull request merged**